Subject: [PATCH] Fix -Wimplicit-function-declaration in configure
---
Index: Zend/Zend.m4
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Zend/Zend.m4 b/Zend/Zend.m4
--- a/Zend/Zend.m4	(revision bf574c2b67a1f786e36cf679f41b758b973a82c4)
+++ b/Zend/Zend.m4	(date 1711832660837)
@@ -7,10 +7,10 @@
 AC_DEFUN([LIBZEND_CHECK_INT_TYPE],[
 AC_MSG_CHECKING(for $1)
 AC_TRY_COMPILE([
-#if HAVE_SYS_TYPES_H  
+#if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
-#if HAVE_INTTYPES_H  
+#if HAVE_INTTYPES_H
 #include <inttypes.h>
 #elif HAVE_STDINT_H
 #include <stdint.h>
@@ -118,16 +118,16 @@
 		double d = (double) LONG_MIN * LONG_MIN + 2e9;
 
 		if ((long) d == 2e9 && (long) -d == -2e9) {
-			exit(0);
+			return 0;
 		}
 	} else if (sizeof(long) == 8) {
 		double correct = 18e18 - ((double) LONG_MIN * -2); /* Subtract ULONG_MAX + 1 */
 
 		if ((long) 18e18 == correct) { /* On 64-bit, only check between LONG_MAX and ULONG_MAX */
-			exit(0);
+			return 0;
 		}
 	}
-	exit(1);
+	return 1;
 }
 ], [
   AC_DEFINE([ZEND_DVAL_TO_LVAL_CAST_OK], 1, [Define if double cast to long preserves least significant bits])
@@ -137,7 +137,7 @@
 ], [
   AC_MSG_RESULT(no)
 ])
-	
+
 ])
 
 AC_DEFUN([LIBZEND_ENABLE_DEBUG],[
@@ -147,7 +147,7 @@
   ZEND_DEBUG=$enableval
 ],[
   ZEND_DEBUG=no
-])  
+])
 
 ])
 
@@ -158,10 +158,10 @@
   ZEND_MAINTAINER_ZTS=$enableval
 ],[
   ZEND_MAINTAINER_ZTS=no
-])  
+])
 
 AC_ARG_ENABLE(inline-optimization,
-[  --disable-inline-optimization 
+[  --disable-inline-optimization
                           If building zend_execute.lo fails, try this switch],[
   ZEND_INLINE_OPTIMIZATION=$enableval
 ],[
@@ -196,7 +196,7 @@
   AC_DEFINE(ZTS,1,[ ])
   CFLAGS="$CFLAGS -DZTS"
   LIBZEND_CPLUSPLUS_CHECKS
-fi  
+fi
 
 changequote({,})
 if test -n "$GCC" && test "$ZEND_INLINE_OPTIMIZATION" != "yes"; then
@@ -249,16 +249,16 @@
   }
 
   fp = fopen("conftest.zend", "w");
-  fprintf(fp, "%d %d\n", ZEND_MM_ALIGNMENT, zeros);  
+  fprintf(fp, "%d %d\n", ZEND_MM_ALIGNMENT, zeros);
   fclose(fp);
 
-  exit(0);
+  return 0;
 }
 ], [
   LIBZEND_MM_ALIGN=`cat conftest.zend | cut -d ' ' -f 1`
   LIBZEND_MM_ALIGN_LOG2=`cat conftest.zend | cut -d ' ' -f 2`
   AC_DEFINE_UNQUOTED(ZEND_MM_ALIGNMENT, $LIBZEND_MM_ALIGN, [ ])
-  AC_DEFINE_UNQUOTED(ZEND_MM_ALIGNMENT_LOG2, $LIBZEND_MM_ALIGN_LOG2, [ ]) 
+  AC_DEFINE_UNQUOTED(ZEND_MM_ALIGNMENT_LOG2, $LIBZEND_MM_ALIGN_LOG2, [ ])
 ], [], [
   dnl cross-compile needs something here
   LIBZEND_MM_ALIGN=8
@@ -321,6 +321,7 @@
 #include <sys/mman.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <unistd.h>
 #ifndef MAP_ANON
 # ifdef MAP_ANONYMOUS
 #  define MAP_ANON MAP_ANONYMOUS
@@ -374,7 +375,7 @@
   ZEND_SIGNALS=$enableval
 ],[
   ZEND_SIGNALS=no
-])  
+])
 
 AC_CHECK_FUNC(sigaction, [
 	AC_DEFINE(HAVE_SIGACTION, 1, [Whether sigaction() is available])
@@ -395,24 +396,24 @@
 
 ])
 
-AC_MSG_CHECKING(whether /dev/urandom exists) 
-if test -r "/dev/urandom" && test -c "/dev/urandom"; then 
+AC_MSG_CHECKING(whether /dev/urandom exists)
+if test -r "/dev/urandom" && test -c "/dev/urandom"; then
   AC_DEFINE([HAVE_DEV_URANDOM], 1, [Define if the target system has /dev/urandom device])
-  AC_MSG_RESULT(yes) 
-else 
-  AC_MSG_RESULT(no) 
+  AC_MSG_RESULT(yes)
+else
+  AC_MSG_RESULT(no)
 fi
 
-AC_MSG_CHECKING(whether /dev/arandom exists) 
-if test -r "/dev/arandom" && test -c "/dev/arandom"; then 
+AC_MSG_CHECKING(whether /dev/arandom exists)
+if test -r "/dev/arandom" && test -c "/dev/arandom"; then
   AC_DEFINE([HAVE_DEV_ARANDOM], 1, [Define if the target system has /dev/arandom device])
-  AC_MSG_RESULT(yes) 
-else 
-  AC_MSG_RESULT(no) 
-fi 
+  AC_MSG_RESULT(yes)
+else
+  AC_MSG_RESULT(no)
+fi
 
 AC_ARG_ENABLE(gcc-global-regs,
-[  --disable-gcc-global-regs 
+[  --disable-gcc-global-regs
                           whether to enable GCC global register variables],[
   ZEND_GCC_GLOBAL_REGS=$enableval
 ],[
Index: Zend/acinclude.m4
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Zend/acinclude.m4 b/Zend/acinclude.m4
--- a/Zend/acinclude.m4	(revision bf574c2b67a1f786e36cf679f41b758b973a82c4)
+++ b/Zend/acinclude.m4	(date 1711832784812)
@@ -7,7 +7,7 @@
   # min: 2.4 (i.e. 204, major * 100 + minor for easier comparison)
   bison_version_min="204"
   # non-working versions, e.g. "3.0 3.2";
-  # remove "none" when introducing the first incompatible bison version an 
+  # remove "none" when introducing the first incompatible bison version an
   # separate any following additions by spaces
   bison_version_exclude=""
 
@@ -67,7 +67,10 @@
 dnl
 AC_DEFUN([AC_ZEND_BROKEN_SPRINTF],[
   AC_CACHE_CHECK(whether sprintf is broken, ac_cv_broken_sprintf,[
-    AC_TRY_RUN([main() {char buf[20];exit(sprintf(buf,"testing 123")!=11); }],[
+    AC_TRY_RUN([
+    #include <stdio.h>
+    int main() { char buf[20]; return sprintf(buf,"testing 123")!=11; }
+    ],[
       ac_cv_broken_sprintf=no
     ],[
       ac_cv_broken_sprintf=yes
@@ -139,7 +142,7 @@
   else
     AC_MSG_RESULT(no)
   fi
-  
+
   AC_MSG_CHECKING([for usable fpsetprec])
   AC_TRY_LINK([
     #include <machine/ieeefp.h>
Index: ext/standard/config.m4
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ext/standard/config.m4 b/ext/standard/config.m4
--- a/ext/standard/config.m4	(revision bf574c2b67a1f786e36cf679f41b758b973a82c4)
+++ b/ext/standard/config.m4	(date 1711833239638)
@@ -7,20 +7,22 @@
 AC_TRY_RUN( [
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 
 int main(int argc, char **argv)
 {
 	char *filename = tmpnam(NULL);
 	char buffer[64];
 	int result = 0;
-	
+
 	FILE *fp = fopen(filename, "wb");
 	if (NULL == fp)
 		return 0;
 	fputs("line 1\n", fp);
 	fputs("line 2\n", fp);
 	fclose(fp);
-	
+
 	fp = fopen(filename, "rb+");
 	if (NULL == fp)
 		return 0;
@@ -58,7 +60,7 @@
     AC_DEFINE(HAVE_CRYPT, 1, [ ])
   ])
 fi
-  
+
 AC_CACHE_CHECK(for standard DES crypt, ac_cv_crypt_des,[
   AC_TRY_RUN([
 #if HAVE_UNISTD_H
@@ -72,9 +74,9 @@
 int main() {
 #if HAVE_CRYPT
 	char *encrypted = crypt("rasmuslerdorf","rl");
-	exit(!encrypted || strcmp(encrypted,"rl.3StKT.4T8M"));
+	return !encrypted || strcmp(encrypted,"rl.3StKT.4T8M");
 #else
-	exit(1);
+	return 1;
 #endif
 }],[
   ac_cv_crypt_des=yes
@@ -86,6 +88,8 @@
 
 AC_CACHE_CHECK(for extended DES crypt, ac_cv_crypt_ext_des,[
   AC_TRY_RUN([
+#include <string.h>
+
 #if HAVE_UNISTD_H
 #include <unistd.h>
 #endif
@@ -97,9 +101,9 @@
 int main() {
 #if HAVE_CRYPT
 	char *encrypted = crypt("rasmuslerdorf","_J9..rasm");
-	exit(!encrypted || strcmp(encrypted,"_J9..rasmBYk8r9AiWNc"));
+	return !encrypted || strcmp(encrypted,"_J9..rasmBYk8r9AiWNc");
 #else
-	exit(1);
+	return 1;
 #endif
 }],[
   ac_cv_crypt_ext_des=yes
@@ -111,6 +115,8 @@
 
 AC_CACHE_CHECK(for MD5 crypt, ac_cv_crypt_md5,[
 AC_TRY_RUN([
+#include <string.h>
+
 #if HAVE_UNISTD_H
 #include <unistd.h>
 #endif
@@ -124,7 +130,7 @@
 	char salt[15], answer[40];
 	char *encrypted;
 
-	salt[0]='$'; salt[1]='1'; salt[2]='$'; 
+	salt[0]='$'; salt[1]='1'; salt[2]='$';
 	salt[3]='r'; salt[4]='a'; salt[5]='s';
 	salt[6]='m'; salt[7]='u'; salt[8]='s';
 	salt[9]='l'; salt[10]='e'; salt[11]='$';
@@ -132,9 +138,9 @@
 	strcpy(answer,salt);
 	strcat(answer,"rISCgZzpwk3UhDidwXvin0");
 	encrypted = crypt("rasmuslerdorf",salt);
-	exit(!encrypted || strcmp(encrypted,answer));
+	return !encrypted || strcmp(encrypted,answer);
 #else
-	exit(1);
+	return 1;
 #endif
 }],[
   ac_cv_crypt_md5=yes
@@ -146,6 +152,8 @@
 
 AC_CACHE_CHECK(for Blowfish crypt, ac_cv_crypt_blowfish,[
 AC_TRY_RUN([
+#include <string.h>
+
 #if HAVE_UNISTD_H
 #include <unistd.h>
 #endif
@@ -164,9 +172,9 @@
 	strcpy(answer,salt);
 	strcpy(&answer[29],"nIdrcHdxcUxWomQX9j6kvERCFjTg7Ra");
 	encrypted = crypt("rasmuslerdorf",salt);
-	exit(!encrypted || strcmp(encrypted,answer));
+	return !encrypted || strcmp(encrypted,answer);
 #else
-	exit(1);
+	return 1;
 #endif
 }],[
   ac_cv_crypt_blowfish=yes
@@ -178,6 +186,8 @@
 
 AC_CACHE_CHECK(for SHA512 crypt, ac_cv_crypt_sha512,[
 AC_TRY_RUN([
+#include <string.h>
+
 #if HAVE_UNISTD_H
 #include <unistd.h>
 #endif
@@ -195,9 +205,9 @@
 	strcpy(answer, salt);
 	strcat(answer, "EeHCRjm0bljalWuALHSTs1NB9ipEiLEXLhYeXdOpx22gmlmVejnVXFhd84cEKbYxCo.XuUTrW.RLraeEnsvWs/");
 	encrypted = crypt("rasmuslerdorf",salt);
-	exit(!encrypted || strcmp(encrypted,answer));
+	return !encrypted || strcmp(encrypted,answer);
 #else
-	exit(1);
+	return 1;
 #endif
 }],[
   ac_cv_crypt_sha512=yes
@@ -260,7 +270,7 @@
     AC_DEFINE([HAVE_ALIGNOF], 1, [whether the compiler supports __alignof__])
   fi
 
-  dnl 
+  dnl
   dnl Check for __attribute__ ((__aligned__)) support in the compiler
   dnl
   AC_CACHE_CHECK(whether the compiler supports aligned attribute, ac_cv_attribute_aligned,[
@@ -275,7 +285,7 @@
   if test "$ac_cv_attribute_aligned" = "yes"; then
     AC_DEFINE([HAVE_ATTRIBUTE_ALIGNED], 1, [whether the compiler supports __attribute__ ((__aligned__))])
   fi
-    
+
 
   AC_DEFINE_UNQUOTED(PHP_USE_PHP_CRYPT_R, 1, [Whether PHP has to use its own crypt_r for blowfish, des, ext des and md5])
   AC_DEFINE_UNQUOTED(PHP_STD_DES_CRYPT, 1, [Whether the system supports standard DES salt])
@@ -321,8 +331,8 @@
     ac_result=0
     ac_crypt_md5=0
   fi
-  AC_DEFINE_UNQUOTED(PHP_MD5_CRYPT, $ac_result, [Whether the system supports MD5 salt])  
-  
+  AC_DEFINE_UNQUOTED(PHP_MD5_CRYPT, $ac_result, [Whether the system supports MD5 salt])
+
   if test "$ac_cv_crypt_sha512" = "yes"; then
     ac_result=1
     ac_crypt_sha512=1
@@ -351,7 +361,7 @@
 dnl as soon as we support C99.
 AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass mempcpy strpncpy)
 AC_CHECK_DECLS([isnan, isinf], [], [], [[#include <math.h>]])
-AC_FUNC_FNMATCH	
+AC_FUNC_FNMATCH
 
 dnl
 dnl Check if there is a support means of creating a new process
Index: configure.in
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/configure.in b/configure.in
--- a/configure.in	(revision bf574c2b67a1f786e36cf679f41b758b973a82c4)
+++ b/configure.in	(date 1711833050524)
@@ -709,6 +709,7 @@
 [AC_TRY_LINK([#include <netdb.h>],
                 [struct addrinfo *g,h;g=&h;getaddrinfo("","",g,&g);], 
   AC_TRY_RUN([
+#include <string.h>
 #include <netdb.h>
 #include <sys/types.h>
 #ifndef AF_INET
@@ -721,11 +722,11 @@
   hints.ai_flags = AI_NUMERICHOST;
 
   if (getaddrinfo("127.0.0.1", 0, &hints, &ai) < 0) {
-    exit(1);
+    return 1;
   }
 
   if (ai == 0) {
-    exit(1);
+    return 1;
   }
 
   pai = ai;
@@ -733,16 +734,16 @@
   while (pai) {
     if (pai->ai_family != AF_INET) {
       /* 127.0.0.1/NUMERICHOST should only resolve ONE way */
-      exit(1);
+      return 1;
     }
     if (pai->ai_addr->sa_family != AF_INET) {
       /* 127.0.0.1/NUMERICHOST should only resolve ONE way */
-      exit(1);
+      return 1;
     }
     pai = pai->ai_next;
   }
   freeaddrinfo(ai);
-  exit(0);
+  return 0;
 }
   ],ac_cv_func_getaddrinfo=yes, ac_cv_func_getaddrinfo=no, ac_cv_func_getaddrinfo=no),
 ac_cv_func_getaddrinfo=no)])
Index: acinclude.m4
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/acinclude.m4 b/acinclude.m4
--- a/acinclude.m4	(revision bf574c2b67a1f786e36cf679f41b758b973a82c4)
+++ b/acinclude.m4	(date 1711832996401)
@@ -12,7 +12,7 @@
 dnl PHP_HELP_SEPARATOR(title)
 dnl
 dnl Adds separator title into the configure --help display.
-dnl 
+dnl
 AC_DEFUN([PHP_HELP_SEPARATOR],[
 AC_ARG_ENABLE([],[
 $1
@@ -148,7 +148,7 @@
 
 dnl
 dnl PHP_GEN_GLOBAL_MAKEFILE
-dnl 
+dnl
 dnl Generates the global makefile.
 dnl
 AC_DEFUN([PHP_GEN_GLOBAL_MAKEFILE],[
@@ -183,10 +183,10 @@
 dnl
 dnl PHP_ADD_SOURCES(source-path, sources [, special-flags [, type]])
 dnl
-dnl Adds sources which are located relative to source-path to the 
-dnl array of type type.  Sources are processed with optional 
+dnl Adds sources which are located relative to source-path to the
+dnl array of type type.  Sources are processed with optional
 dnl special-flags which are passed to the compiler.  Sources
-dnl can be either written in C or C++ (filenames shall end in .c 
+dnl can be either written in C or C++ (filenames shall end in .c
 dnl or .cpp, respectively).
 dnl
 dnl Note: If source-path begins with a "/", the "/" is removed and
@@ -227,7 +227,7 @@
 dnl shared objects will be built from the sources.
 dnl
 dnl Should not be used directly.
-dnl 
+dnl
 AC_DEFUN([PHP_ADD_SOURCES_X],[
 dnl relative to source- or build-directory?
 dnl ac_srcdir/ac_bdir include trailing slash
@@ -236,20 +236,20 @@
   /*[)] ac_srcdir=`echo "$1"|cut -c 2-`"/"; ac_bdir=$ac_srcdir; ac_inc="-I$ac_bdir -I$abs_srcdir/$ac_bdir" ;;
   *[)] ac_srcdir="$abs_srcdir/$1/"; ac_bdir="$1/"; ac_inc="-I$ac_bdir -I$ac_srcdir" ;;
   esac
-  
+
 dnl how to build .. shared or static?
   ifelse($5,yes,_PHP_ASSIGN_BUILD_VARS(shared),_PHP_ASSIGN_BUILD_VARS(php))
 
 dnl iterate over the sources
   old_IFS=[$]IFS
   for ac_src in $2; do
-  
+
 dnl remove the suffix
       IFS=.
       set $ac_src
       ac_obj=[$]1
       IFS=$old_IFS
-      
+
 dnl append to the array which has been dynamically chosen at m4 time
       $4="[$]$4 [$]ac_bdir[$]ac_obj.lo"
 
@@ -364,7 +364,7 @@
 ])
 
 dnl -------------------------------------------------------------------------
-dnl Macros to modify LIBS, INCLUDES, etc. variables 
+dnl Macros to modify LIBS, INCLUDES, etc. variables
 dnl -------------------------------------------------------------------------
 
 dnl
@@ -486,7 +486,7 @@
 dnl
 dnl PHP_ADD_INCLUDE(path [,before])
 dnl
-dnl add an include path. 
+dnl add an include path.
 dnl if before is 1, add in the beginning of INCLUDES.
 dnl
 AC_DEFUN([PHP_ADD_INCLUDE],[
@@ -774,7 +774,7 @@
   PHP_BUILD_PROGRAM
   OVERALL_TARGET=libphp[]$PHP_MAJOR_VERSION[.la]
   php_sapi_module=shared
-  
+
   php_c_pre=$shared_c_pre
   php_c_meta=$shared_c_meta
   php_c_post=$shared_c_post
@@ -889,7 +889,7 @@
 ])
   else
     PHP_SAPI=$1
-  fi  
+  fi
 
   PHP_ADD_BUILD_DIR([sapi/$1])
 
@@ -946,7 +946,7 @@
 dnl "shared" can be set to "shared" or "yes" to build the extension as
 dnl a dynamically loadable library. Optional parameter "sapi_class" can
 dnl be set to "cli" to mark extension build only with CLI or CGI sapi's.
-dnl "extra-cflags" are passed to the compiler, with 
+dnl "extra-cflags" are passed to the compiler, with
 dnl @ext_srcdir@ and @ext_builddir@ being substituted.
 dnl "cxx" can be used to indicate that a C++ shared module is desired.
 dnl "zend_ext" indicates a zend extension.
@@ -999,7 +999,7 @@
   PHP_ADD_BUILD_DIR($ext_builddir)
 
 dnl Set for phpize builds only
-dnl --------------------------- 
+dnl ---------------------------
   if test "$ext_builddir" = "."; then
     PHP_PECL_EXTENSION=$1
     PHP_SUBST(PHP_PECL_EXTENSION)
@@ -1029,7 +1029,7 @@
 dnl default is false and should halt the build.
 dnl To be effective, this macro must be invoked *after* PHP_NEW_EXTENSION.
 dnl The extension on which it depends must also have been configured.
-dnl See ADD_EXTENSION_DEP in win32 build 
+dnl See ADD_EXTENSION_DEP in win32 build
 dnl
 AC_DEFUN([PHP_ADD_EXTENSION_DEP], [
   am_i_shared=$[PHP_]translit($1,a-z_-,A-Z__)[_SHARED]
@@ -1191,7 +1191,7 @@
   struct tm t, *s;
   time_t old = 0;
   char buf[27], *p;
-  
+
   s = gmtime_r(&old, &t);
   p = asctime_r(&t, buf, 26);
   if (p == buf && s == &t) return (0);
@@ -1225,14 +1225,14 @@
 #include <unistd.h>
 #include <errno.h>
 $1
-    main() {
+    int main() {
     int fd = open("conftest_in", O_WRONLY|O_CREAT, 0600);
 
-    if (fd < 0) exit(1);
-    if (pwrite(fd, "text", 4, 0) != 4) exit(1);
+    if (fd < 0) return 1;
+    if (pwrite(fd, "text", 4, 0) != 4) return 1;
     /* Linux glibc breakage until 2.2.5 */
-    if (pwrite(fd, "text", 4, -1) != -1 || errno != EINVAL) exit(1);
-    exit(0);
+    if (pwrite(fd, "text", 4, -1) != -1 || errno != EINVAL) return 1;
+    return 0;
     }
 
   ],[
@@ -1255,14 +1255,14 @@
 #include <unistd.h>
 #include <errno.h>
 $1
-    main() {
-    char buf[3]; 
+    int main() {
+    char buf[3];
     int fd = open("conftest_in", O_RDONLY);
-    if (fd < 0) exit(1);
-    if (pread(fd, buf, 2, 0) != 2) exit(1);
+    if (fd < 0) return 1;
+    if (pread(fd, buf, 2, 0) != 2) return 1;
     /* Linux glibc breakage until 2.2.5 */
-    if (pread(fd, buf, 2, -1) != -1 || errno != EINVAL) exit(1);
-    exit(0);
+    if (pread(fd, buf, 2, -1) != -1 || errno != EINVAL) return 1;
+    return 0;
     }
   ],[
     ac_cv_pread=yes
@@ -1293,7 +1293,7 @@
     if test "$ac_cv_pwrite" = "64"; then
       AC_DEFINE(PHP_PWRITE_64, 1, [whether pwrite64 is default])
     fi
-  fi  
+  fi
 ])
 
 dnl
@@ -1315,7 +1315,7 @@
     if test "$ac_cv_pread" = "64"; then
       AC_DEFINE(PHP_PREAD_64, 1, [whether pread64 is default])
     fi
-  fi  
+  fi
 ])
 
 dnl
@@ -1353,7 +1353,7 @@
 
 dnl
 dnl PHP_READDIR_R_TYPE
-dnl 
+dnl
 AC_DEFUN([PHP_READDIR_R_TYPE],[
   dnl HAVE_READDIR_R is also defined by libmysql
   AC_CHECK_FUNC(readdir_r,ac_cv_func_readdir_r=yes,ac_cv_func_readdir=no)
@@ -1363,25 +1363,26 @@
 #define _REENTRANT
 #include <sys/types.h>
 #include <dirent.h>
+#include <unistd.h>
 
 #ifndef PATH_MAX
 #define PATH_MAX 1024
 #endif
 
-main() {
+int main() {
   DIR *dir;
   char entry[sizeof(struct dirent)+PATH_MAX];
   struct dirent *pentry = (struct dirent *) &entry;
 
   dir = opendir("/");
-  if (!dir) 
-    exit(1);
+  if (!dir)
+    return 1;
   if (readdir_r(dir, (struct dirent *) entry, &pentry) == 0) {
     close(dir);
-    exit(0);
+    return 0;
   }
   close(dir);
-  exit(1);
+  return 1;
 }
     ],[
       ac_cv_what_readdir_r=POSIX
@@ -1411,7 +1412,7 @@
 
 dnl
 dnl PHP_TM_GMTOFF
-dnl 
+dnl
 AC_DEFUN([PHP_TM_GMTOFF],[
 AC_CACHE_CHECK([for tm_gmtoff in struct tm], ac_cv_struct_tm_gmtoff,
 [AC_TRY_COMPILE([#include <sys/types.h>
@@ -1514,7 +1515,7 @@
 main() {
   char buf[20];
   int res = 0;
-  res = res || (snprintf(buf, 2, "marcus") != 6); 
+  res = res || (snprintf(buf, 2, "marcus") != 6);
   res = res || (buf[1] != '\0');
   /* Implementations may consider this as an encoding error */
   snprintf(buf, 0, "boerger");
@@ -1522,7 +1523,7 @@
   res = res || (buf[0] != 'm');
   res = res || (snprintf(NULL, 0, "boerger") != 7);
   res = res || (snprintf(buf, sizeof(buf), "%f", 0.12345678) != 8);
-  exit(res); 
+  exit(res);
 }
     ],[
       ac_cv_broken_snprintf=no
@@ -1629,7 +1630,7 @@
   if test "$ac_cv_sockaddr_storage" = "yes"; then
     AC_DEFINE(HAVE_SOCKADDR_STORAGE, 1, [Whether you have struct sockaddr_storage])
   fi
-  dnl Check if field sa_len exists in struct sockaddr 
+  dnl Check if field sa_len exists in struct sockaddr
   AC_CACHE_CHECK([for field sa_len in struct sockaddr],ac_cv_sockaddr_sa_len,[
     AC_TRY_COMPILE([#include <sys/types.h>
 #include <sys/socket.h>],
@@ -1670,9 +1671,9 @@
 AC_DEFUN([PHP_EBCDIC], [
   AC_CACHE_CHECK([whether system uses EBCDIC],ac_cv_ebcdic,[
   AC_TRY_RUN( [
-int main(void) { 
-  return (unsigned char)'A' != (unsigned char)0xC1; 
-} 
+int main(void) {
+  return (unsigned char)'A' != (unsigned char)0xC1;
+}
 ],[
   ac_cv_ebcdic=yes
 ],[
@@ -1711,16 +1712,17 @@
   AC_CACHE_VAL(_cv_have_broken_glibc_fopen_append,[
   AC_TRY_RUN([
 #include <stdio.h>
+#include <unistd.h>
 int main(int argc, char *argv[])
 {
   FILE *fp;
   long position;
   char *filename = tmpnam(NULL);
-  
+
   fp = fopen(filename, "w");
   if (fp == NULL) {
     perror("fopen");
-    exit(2);
+    return 2;
   }
   fputs("foobar", fp);
   fclose(fp);
@@ -1794,13 +1796,13 @@
 
 cookie_io_functions_t funcs = {reader, writer, seeker, closer};
 
-main() {
+int main() {
   struct cookiedata g = { 0 };
   FILE *fp = fopencookie(&g, "r", funcs);
 
   if (fp && fseek(fp, 8192, SEEK_SET) == 0 && g.pos == 8192)
-    exit(0);
-  exit(1);
+    return 0;
+  return 1;
 }
 
 ], [
@@ -1810,7 +1812,7 @@
 ], [
   cookie_io_functions_use_off64_t=no
 ])
-    
+
     else
 
 dnl older glibc versions (up to 2.1.2 ?)
@@ -1847,7 +1849,7 @@
 AC_DEFUN([PHP_CHECK_LIBRARY], [
   save_old_LDFLAGS=$LDFLAGS
   ac_stuff="$5"
-  
+
   save_ext_shared=$ext_shared
   ext_shared=yes
   PHP_EVAL_LIBLINE([$]ac_stuff, LDFLAGS)
@@ -1937,11 +1939,11 @@
   unset ac_cv_func_$1
   unset ac_cv_func___$1
   unset found
-  
+
   AC_CHECK_FUNC($1, [found=yes],[ AC_CHECK_FUNC(__$1,[found=yes],[found=no]) ])
 
   case $found in
-  yes[)] 
+  yes[)]
     PHP_DEF_HAVE($1)
     ac_cv_func_$1=yes
   ;;
@@ -2012,7 +2014,7 @@
 dnl PHP_CHECK_64BIT([do if 32], [do if 64])
 dnl
 dnl This macro is used to detect if we're at 64-bit platform or not.
-dnl It could be useful for those external libs, that have different precompiled 
+dnl It could be useful for those external libs, that have different precompiled
 dnl versions in different directories.
 dnl
 AC_DEFUN([PHP_CHECK_64BIT],[
@@ -2082,9 +2084,9 @@
     *mawk)
       AC_MSG_WARN([mawk is known to have problems on some systems. You should install GNU awk])
       ;;
-    *gawk)  
+    *gawk)
       ;;
-    bork)   
+    bork)
       AC_MSG_ERROR([Could not find awk; Install GNU awk])
       ;;
     *)
@@ -2119,7 +2121,7 @@
 AC_DEFUN([PHP_PROG_LEX], [
 dnl we only support certain flex versions
   flex_version_list="2.5.4"
-   
+
   AC_PROG_LEX
   if test "$LEX" = "flex"; then
 dnl AC_DECL_YYTEXT is obsolete since autoconf 2.50 and merged into AC_PROG_LEX
@@ -2149,7 +2151,7 @@
   else
     flex_version=none
   fi
-  
+
   case $php_cv_flex_version in
     ""|invalid[)]
       if test -f "$abs_srcdir/Zend/zend_language_scanner.c" && test -f "$abs_srcdir/Zend/zend_ini_scanner.c"; then
@@ -2183,7 +2185,7 @@
         php_cv_re2c_version=invalid
       else
         php_cv_re2c_version="`$RE2C --version | cut -d ' ' -f 2  2>/dev/null` (ok)"
-      fi 
+      fi
     ])
   fi
   case $php_cv_re2c_version in
@@ -2309,7 +2311,7 @@
   fi
 ])
 
-dnl 
+dnl
 dnl PHP_SETUP_OPENSSL(shared-add [, action-found [, action-not-found]])
 dnl
 dnl Common setup macro for openssl
@@ -2356,7 +2358,7 @@
 
   dnl If pkg-config fails for some reason, revert to the old method
   if test "$found_openssl" = "no"; then
-  
+
     if test "$PHP_OPENSSL_DIR" = "yes"; then
       PHP_OPENSSL_DIR="/usr/local/ssl /usr/local /usr /usr/local/openssl"
     fi
@@ -2395,7 +2397,7 @@
     CPPFLAGS=$old_CPPFLAGS
 
     PHP_ADD_INCLUDE($OPENSSL_INCDIR)
-  
+
     PHP_CHECK_LIBRARY(crypto, CRYPTO_free, [
       PHP_ADD_LIBRARY(crypto,,$1)
     ],[
@@ -2430,7 +2432,7 @@
   fi
 ])
 
-dnl 
+dnl
 dnl PHP_SETUP_ICONV(shared-add [, action-found [, action-not-found]])
 dnl
 dnl Common setup macro for iconv
@@ -2494,7 +2496,7 @@
     if test -z "$ICONV_DIR"; then
       AC_MSG_ERROR([Please specify the install prefix of iconv with --with-iconv=<DIR>])
     fi
-  
+
     if test -f $ICONV_DIR/$PHP_LIBDIR/lib$iconv_lib_name.a ||
        test -f $ICONV_DIR/$PHP_LIBDIR/lib$iconv_lib_name.$SHLIB_SUFFIX_NAME
     then
@@ -2528,7 +2530,7 @@
   fi
 ])
 
-dnl 
+dnl
 dnl PHP_SETUP_LIBXML(shared-add [, action-found [, action-not-found]])
 dnl
 dnl Common setup macro for libxml
@@ -2585,7 +2587,7 @@
 dnl Misc. macros
 dnl -------------------------------------------------------------------------
 
-dnl 
+dnl
 dnl PHP_INSTALL_HEADERS(path [, file ...])
 dnl
 dnl PHP header files to be installed
@@ -2596,7 +2598,7 @@
       PHP_RUN_ONCE(INSTALLHEADERS, $header_file, [
         INSTALL_HEADERS="$INSTALL_HEADERS $header_file"
       ])
-    done 
+    done
   ], [
     header_path=$1
     for header_file in $2; do
@@ -2604,7 +2606,7 @@
       PHP_RUN_ONCE(INSTALLHEADERS, $hp_hf, [
         INSTALL_HEADERS="$INSTALL_HEADERS $hp_hf"
       ])
-    done 
+    done
   ])
 ])
 
@@ -2627,7 +2629,7 @@
 
 dnl
 dnl PHP_DEBUG_MACRO(filename)
-dnl 
+dnl
 AC_DEFUN([PHP_DEBUG_MACRO],[
   DEBUG_LOG=$1
   cat >$1 <<X
@@ -2689,7 +2691,7 @@
   echo "'[$]0' \\" >> $1
   if test `expr " [$]0" : " '.*"` = 0; then
     CONFIGURE_COMMAND="$CONFIGURE_COMMAND '[$]0'"
-  else 
+  else
     CONFIGURE_COMMAND="$CONFIGURE_COMMAND [$]0"
   fi
   CONFIGURE_ARGS="$clean_configure_args"
@@ -2757,7 +2759,7 @@
         if test "$PHP_MAJOR_VERSION" -lt "6"; then
           case $arg_name in
             enable-zend-multibyte[)] continue;;
-          esac 
+          esac
         fi
 
         is_arg_set=php_[]`echo [$]arg_name | tr 'ABCDEFGHIJKLMNOPQRSTUVWXYZ-' 'abcdefghijklmnopqrstuvwxyz_'`
@@ -2839,7 +2841,7 @@
 ],[
 CRYPTD buffer;
 crypt_r("passwd", "hash", &buffer);
-], 
+],
 php_cv_crypt_r_style=cryptd)
 
     if test "$php_cv_crypt_r_style" = "none"; then
@@ -2849,7 +2851,7 @@
 ],[
 struct crypt_data buffer;
 crypt_r("passwd", "hash", &buffer);
-], 
+],
 php_cv_crypt_r_style=struct_crypt_data)
     fi
 
@@ -2861,7 +2863,7 @@
 ],[
 struct crypt_data buffer;
 crypt_r("passwd", "hash", &buffer);
-], 
+],
 php_cv_crypt_r_style=struct_crypt_data_gnu_source)
     fi
     ])
